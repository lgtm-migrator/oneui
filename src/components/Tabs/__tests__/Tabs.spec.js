import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render } from '@testing-library/react';
import Tabs from '../Tabs';
import Tab from '../Tab';

describe('<Tabs> that renders tab container with some tabs', () => {
    it('should render simple tabs', () => {
        const { container } = render(
            <Tabs activeTabId="1">
                <Tab id="1" label="First">
                    Some content
                </Tab>
                <Tab id="2" label="Second">
                    Some more content
                </Tab>
            </Tabs>
        );
        expect(container).toMatchSnapshot();
    });

    it('should switch tabs correctly', () => {
        const onChange = jest.fn();

        const { container } = render(
            <Tabs activeTabId="1" onChange={onChange}>
                <Tab id="1" label="First">
                    Some content
                </Tab>
                <Tab id="2" label="Second">
                    Some more content
                </Tab>
            </Tabs>
        );

        container
            .find('TabMenu')
            .find('TabItem')
            .at(1)
            .simulate('click');
        expect(container.activeTabId).toBe('2');
        expect(onChange).toHaveBeenCalledWith('2');
    });

    it('should switch tabs correctly (programmatically)', () => {
        const wrapper = mount(
            <Tabs activeTabId="1">
                <Tab id="1" label="First">
                    Some content
                </Tab>
                <Tab id="2" label="Second">
                    Some more content
                </Tab>
            </Tabs>
        );
        wrapper.setProps(
            {
                activeTabId: '2',
            },
            () => {
                expect(wrapper.state().activeTabId).toBe('2');
                expect(wrapper.state().derivedTabId).toBe('2');
                expect(wrapper.find('TabContent').text()).toEqual('Some more content');
            }
        );
    });
});
