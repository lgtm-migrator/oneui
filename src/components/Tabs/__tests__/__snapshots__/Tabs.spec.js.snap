// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs> that renders tab container with some tabs should render simple tabs 1`] = `
<div>
  <div
    class="TabMenu"
    role="tablist"
  >
    <a
      aria-selected="true"
      class="TabItem TabItem--isActive"
      href="#"
      id="1"
      role="tab"
      tabindex="0"
    >
      First
    </a>
    <a
      aria-selected="false"
      class="TabItem"
      href="#"
      id="2"
      role="tab"
      tabindex="0"
    >
      Second
    </a>
  </div>
  <div
    aria-labelledby="1"
    class="TabContent"
    role="tabpanel"
  >
    Some content
  </div>
</div>
`;
